src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -358,6 +358,29 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         :param inputs: the input schema for the pipeline
         :return: the inputs for the engine
         """
+        if not self.cache_support_enabled and inputs.max_tokens > 1:
+            raise ValueError(
+                "The model used for inference does not support kv cache. It is "
+                "assumed that it maps from the token sequence to predicted logits."
+                "Set `max_tokens` to 1 to support that scenario."
+            )
+
+        if inputs.num_generated_predictions > 1:
```

Comment:
could you add a short comment that briefly describes what happens here (may not be obvious to somebody who picks up the code later)

Reply:
yes will add. much of the code is if the user provides the same prompt as an input multiple times (as seems to be a common testing case). For that case, we ignore `num_generated_predictions` but we can also just treat them the same way.